tests/kernel/context/src/main.c

```diff
@@ -224,20 +227,30 @@ static void _test_kernel_cpu_idle(int atomic)
 	unsigned int i, key;
 	uint32_t dur = k_ms_to_ticks_ceil32(10);
 	uint32_t slop = 1 + k_ms_to_ticks_ceil32(1);
+	int idle_loops;
+/* Something else may be running in the background waking the CPU, but we still want to
+ * check that k_cpu_idle() idles the CPU and we are not just busy waiting. We allow the loop
+ * to run up to 128 times in 10ms, which allows for plenty of other systems' interrupts
+ */
+#define MAX_IDLE_WAKES 128
```

Comment:
This threshold seems too high to me. If you handed me a device that was waking up spuriously every 40ms (5s/128) and asked me if its CPU idle implementation was working, I'd probably say "hell no". :)

Maybe limit this (i.e. set the max to zero by default) to specific platforms/configurations (just nRF with SYNC_RTC, right?) that are known to have internal idle wakeups, and tune the number as tight as we can?

Reply:
Well.. it could be anything that happens to run at the same time as the test. And as the test runs very soon after boot, there may be quite a few platform components doing something.
128 just felt like safe enough to avoid false negatives (unless the platform is, say, printing thru the UART and interrupt per byte or something like that) while at the same time, still checking that we are not just busy waiting.
I can certainly lower it to, say, 16 or so, which should cover most platforms.
Does that sound good to you?

Reply:
Right, but that's a software robustness argument.  This is a test, and the assumption in the test is clearly that "nothing should wake us up".  It turns out there's an edge case where something might, so (if we don't want to fix the edge case) let's tune the detection to just that edge case.

Again, if this test is waking up on any platform except nRF+SYNC_RTC, that's a failure we want to know about.  "Too much fiddling" is a very minor sin when compared to "we lost test coverage". :)